determined-ai/determined
# chore: replace router hooks for react-router-dom v6

## Motivation

- In order to update React to v18, we need to update React-Router-Dom as well.
- Some Jira tickets demand React18 
  - https://determinedai.atlassian.net/browse/DET-7178
  - https://determinedai.atlassian.net/browse/DET-8046


## Description

Previous: #4788
Followed [the instruction](https://github.com/remix-run/react-router/discussions/8753)

What I have done in this PR
- replace useHistory with useNavigate
- use new `useParams`
  - Replaced `interface` with `type` to adjust new react-router-dom
- Fix test cases

## Plan


```mermaid
gantt
    title Update React-Router-Dom Plan
    dateFormat  YYYY-MM-DD
    section React-Router v6
    Replace hooks           : crit, a1, 2022-09-28, 3d
    Replace Root             :a2, after a1 , 5d
    Wrap up React Router v6         :a3, after a2 , 3d
    section React v18
    Update to React v18      :b1,  2022-10-12, 5d
```




## Test Plan
- Check if routing works as expected
- Check if params works as expected
- Check if location works as expected



## Commentary (optional)




## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.





[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
